### PR TITLE
Support UWP build with CMakePresets.json

### PR DESCRIPTION
**Pull request by luncliff**
### Changes

* Allow `WindowsStore` for `CMAKE_SYSTEM_NAME`
* Create `CMakePresets.json` that holds UWP build settings
    1. WindowsStore + SDK 10.0 + x86_64 (Debug/Release)
    2. WindowsStore + SDK 10.0 + ARM64 (Debug/Release)

#### GitHub Actions

* New build job `cmake-uwp` with [lukka/run-cmake@v10](https://github.com/lukka/run-cmake/releases/tag/v10.0)

#### Warnings

Remove uninitialized variable compiler warning in src/x86/name.c

### CODE REVIEW

**Review by Maratyszcza — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -171,7 +171,7 @@ IF(CPUINFO_SUPPORTED_PLATFORM)
       LIST(APPEND CPUINFO_SRCS
         src/arm/android/properties.c)
     ENDIF()
-  ELSEIF(CMAKE_SYSTEM_NAME STREQUAL "Windows" AND CPUINFO_TARGET_PROCESSOR STREQUAL "ARM64")
+  ELSEIF((CMAKE_SYSTEM_NAME MATCHES "Windows") AND CPUINFO_TARGET_PROCESSOR STREQUAL "ARM64")
```
Comment (Maratyszcza) [inline CMakeLists.txt:174]: Should be "^Windows"

File: CMakeLists.txt [hunk 2]
```diff
@@ -211,10 +211,12 @@ ADD_LIBRARY(cpuinfo_internals STATIC ${CPUINFO_SRCS})
...
 CPUINFO_TARGET_RUNTIME_LIBRARY(cpuinfo)
-IF(CMAKE_SYSTEM_NAME MATCHES "^(Windows|CYGWIN|MSYS)$")
+IF(CMAKE_SYSTEM_NAME MATCHES "^(Windows|WindowsStore|CYGWIN|MSYS)$")
   # Target Windows 7+ API
-  TARGET_COMPILE_DEFINITIONS(cpuinfo PRIVATE _WIN32_WINNT=0x0601)
-  TARGET_COMPILE_DEFINITIONS(cpuinfo_internals PRIVATE _WIN32_WINNT=0x0601)
+  TARGET_COMPILE_DEFINITIONS(cpuinfo PRIVATE _WIN32_WINNT=0x0601 _CRT_SECURE_NO_WARNINGS)
+  TARGET_COMPILE_DEFINITIONS(cpuinfo_internals PRIVATE _WIN32_WINNT=0x0601 _CRT_SECURE_NO_WARNINGS)
+  # Explicitly link Kernel32 for UWP build
+  TARGET_LINK_LIBRARIES(cpuinfo PUBLIC Kernel32)
```
Comment (Maratyszcza) [inline CMakeLists.txt:219]: Does this line still work with Windows/CYGWIN/MSYS?

File: CMakeLists.txt (hunk 1 above)
Comment (luncliff) [reply CMakeLists.txt]: Nice. Fixed in e07e98b923d0e566a5a468d1cf3fd1c06c010757

File: CMakeLists.txt (hunk 2 above)
Comment (luncliff) [reply CMakeLists.txt]: Not sure... Made a guard with "WindowsStore" check.

https://github.com/pytorch/cpuinfo/blob/e07e98b923d0e566a5a468d1cf3fd1c06c010757/CMakeLists.txt#L218-L221

**Review by Maratyszcza — APPROVED** [review]

**Pull request merged**